### ISSUE TITLE
Enable loading edited csv schemas

### ISSUE THREAD

**Issue by JohannaOtt** [halestudio/hale#782]
When using a csv file as schema and data source and adding columns to it, hale studio could automatically recognize that there are additional columns in the csv file and provide a dialogue to the customer in order to define the new columns' datatypes.
This would facilitate the use of csv files, espeacially if those are edited from time to time and these changes need to be addresses in the alignment.

**Comment by thorsten-reitz** [comment]
Is it possible to use the "reload and update schemas" process for CSVs? Because that would normally be the way to deal with changes to the schemas. hale studio doesn't generally recognize such changes.

**Comment by thorsten-reitz** [comment]
@JohannaOtt Do you think we should maintain this ticket?

**Comment by JohannaOtt** [comment]
Yes I assume so. At least I am not missing it regularly. Maybe @florianesser and @annat2022 can comment. AFAIK you are using csv schemas more often?

**Comment by thorsten-reitz** [comment]
I did use more CSV schemas, but am normally just reloading the schema through the respective process if I need to make a change.

**Comment by annat2022** [comment]
> I did use more CSV schemas, but am normally just reloading the schema through the respective process if I need to make a change.

I'm not using csv schemas often. But I just tested it:
If I import a csv-file as schema with two attribute fields, save the hale studio project, adjust the used csv-file and add a third field, reload the schema or reload the hale studio project then the third attribute is not added and I'm getting an error for the csv-file import.

![grafik](https://github.com/halestudio/hale/assets/116007248/f1b847f5-6ce7-4c6d-85f6-12fed4c5a9f7)

Might be that I'm missing something. But I assume in this case it would be definitely good when hale studio could handle these changes.
